src/Symfony/Component/Messenger/MessageRecorder.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger;
+
+use Symfony\Component\Cache\ResettableInterface;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ * @author Matthias Noback <matthiasnoback@gmail.com>
+ */
+class MessageRecorder implements MessageRecorderInterface, RecordedMessageContainerInterface, ResettableInterface
+{
+    private $messages = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRecordedMessages(): array
+    {
+        return $this->messages;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function resetRecordedMessages(): void
+    {
+        $this->reset();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function reset()
```

Comment:
Why do you keep this method? You should remove it. Like I said, `reset`does not mean anything for a domain object.

Reply:
I agree. But I need this service to implement `ResettableInterface`. You do not need to implement this method in your domain objects. I removed it from the interface.  =)

Reply:
ok :+1: